test/distributed-ranges/shp/CMakeLists.txt

```diff
@@ -76,7 +77,10 @@ add_executable(
 add_executable(shp-tests-3 shp-tests.cpp containers-3.cpp copy-3.cpp)
 
 foreach(test-exec IN ITEMS shp-tests shp-tests-3)
-  target_link_libraries(${test-exec} PRIVATE GTest::gtest_main fmt::fmt oneDPL)
+  target_link_libraries(${test-exec} PRIVATE GTest::gtest_main oneDPL)
+  if(fmt_FOUND)
+    target_link_libraries(${test-exec} PRIVATE fmt::fmt)
```

Comment:
e.g. just write here `target_compile_options(${test-exec} PRIVATE -DUSE_FMT)` and we are done with one line and no extra variables.